corehq/apps/hqwebapp/models.py

```diff
@@ -53,3 +55,26 @@ def get_latest_alert(cls):
             return active_alerts[0]
         else:
             return ''
+
+
+@architect.install('partition', type='range', subtype='date', constraint='month', column='timestamp')
+class UserAccessLog(models.Model):
+    TYPE_LOGIN = 'login'
+    TYPE_LOGOUT = 'logout'
+    TYPE_FAILURE = 'failure'
+
+    ACTIONS = (
+        (TYPE_LOGIN, 'Login'),
+        (TYPE_LOGOUT, 'Logout'),
+        (TYPE_FAILURE, 'Login Failure')
+    )
+
+    user_id = models.CharField(max_length=255, db_index=True)
+    action = models.CharField(max_length=20, choices=ACTIONS)
+    ip = models.GenericIPAddressField()
+    user_agent = models.CharField(max_length=255, blank=True)
```

Comment:
This doesn't seem long enough to store every user agent string. Also, not sure it's a great idea to store full user agent string in every row here because of the amount of duplication. I like the ideas in [this so answer](https://stackoverflow.com/a/13210391/10840) concerning how to store them.

Reply:
Hadn't thought about this, will implement one of these solutions